pkg/operator/ceph/file/mds/mds.go

```diff
@@ -149,10 +155,11 @@ func (c *Cluster) Start() error {
 				return fmt.Errorf("failed to get existing mds deployment %s for update: %+v", d.Name, err)
 			}
 		}
-		// create unique key for each mds saved to k8s secret
-		if err := c.generateKeyring(mdsConfig, createdDeployment.UID); err != nil {
-			return fmt.Errorf("failed to generate keyring for %s. %+v", resourceName, err)
+
+		if err := c.associateKeyring(keyring, createdDeployment); err != nil {
+			return fmt.Errorf("failed to associate keyring with deployment for %q. %+v", resourceName, err)
```

Comment:
Ditto this comment. https://github.com/rook/rook/pull/4235#discussion_r342193675